### DISCUSSION TITLE
How do I add raise_fault_policy in apiman 3.1.1

### DISCUSSION THREAD

**Discussion by zumikavvayi** [apiman/apiman#2508] [Help] [answered]
I've designed a custom event policy that checks certain conditions and publish events. However, if there is an exception during the event, I want the request to fail so that the client can be alerted. I noticed some old documentation mentioning a "raise_fault_policy" that might help with this issue. How can I accomplish this in apiman 3.1.1?

**Comment by msavy** [comment, marked as answer]
`chain.doFailure` 

Example:

```java
IPolicyFailureFactoryComponent failureFactory = context.getComponent(IPolicyFailureFactoryComponent.class);
PolicyFailure failure = failureFactory.createFailure(...);
chain.doFailure(failure);
```

**Reply by ronimhd** [reply]
Can we call that doFailure code/logic in IDataPolicy implementation? Let say if there is exception in getRequestDataHandler part.

**Reply by msavy** [reply]
You can try, but it might not work depending how far along the request/response is. You may need to use an alternative method to terminate the connection.
